src/checks/json-path-errors.ts

```diff
@@ -0,0 +1,28 @@
+import { JSONPath } from 'jsonpath-plus';
+import { StateMachineDefinition, StateMachineError, StateMachineErrorCode } from '../types';
+
+export default function jsonPathErrors(definition: StateMachineDefinition): StateMachineError[] {
+  return JSONPath({ json: definition, path: '$..[InputPath,OutputPath,ResultPath]' })
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .filter((path: any) => typeof path === 'string')
+    .map((path: string) => {
+      // Context object path starts with $$
+      // https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html#contextobject-access
+      if (path.startsWith('$$')) {
+        return path.substring(1);
+      }
+      return path;
+    })
```

Comment:
since this validator is running after schema validation, you know that AJV already ensured that the value of every **InputPath**, **OutputPath**, and **ResultPath** is a `string`.

It's safe to do the following:

```typescript
const paths : string[] = JSONPath( {.....});
```

Reply:
also note that the "$$" rewrite above may need to occur in multiple spots if the path is an intrinsic function with args like: `States.Format('pattern {} {}', $$.prop1, $$.prop2)`

this is where the asl-path-validator work would help.

Reply:
Right!